contracts/wawaka/attestation-test/attestation-test.cpp

```diff
@@ -361,6 +361,37 @@ bool reveal_secret(const Message& msg, const Environment& env, Response& rsp)
     return rsp.value(result, false);
 }
 
+// XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
+// NAME: verify_sgx_report
+// XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
+#define VERIFY_SGX_REPORT_PARAMETER_SCHEMA      \
+     "{"                                        \
+          "\"certificate\":\"\","               \
+          "\"report\":\"\","                    \
+          "\"signature\":\"\""                  \
+     "}"
+
+bool verify_sgx_report(const Message& msg, const Environment& env, Response& rsp)
+{
+    ASSERT_SUCCESS(rsp, msg.validate_schema(VERIFY_SGX_REPORT_PARAMETER_SCHEMA),
+                   "invalid request, missing required parameters");
+
+    const std::string certificate(msg.get_string("certificate"));
+    const std::string report(msg.get_string("report"));
+    const std::string signature(msg.get_string("signature"));
+
+    CONTRACT_SAFE_LOG(3, "report: %s", report.c_str());
+
+    int status = verify_sgx_report(certificate.c_str(), certificate.length(),
+                                   report.c_str(), report.length(),
+                                   signature.c_str(), signature.length());
+
+    ww::value::Boolean result(status == 0);
```

Comment:
This correctly returns `true` on successful verification, and `false` on insuccessful verification, but incorrectly returns `true` on error.

Reply:
actually status returns 0 (verify_success) on success. that's why this should succeed.

Reply:
Fixed.